llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -5001,6 +5019,10 @@ LoopVectorizationCostModel::selectInterleaveCount(ElementCount VF,
   if (!Legal->isSafeForAnyVectorWidth())
     return 1;
 
+  // We don't attempt to perform interleaving for early exit loops.
+  if (Legal->hasUncountableEarlyExit())
+    return 1;
```

Comment:
Could you add more details on why not? Is all that would be needed to deal with multiple parts in AnyOf & co?